build.gradle

```diff
@@ -1,104 +1,124 @@
-    }
-}
-
-task javadocJar(type: Jar) {
-    classifier = 'javadoc'
-    from javadoc
-}
-
-task sourcesJar(type: Jar) {
-    classifier = 'sources'
-    from sourceSets.main.allSource
-}
-
-artifacts {
-    archives javadocJar, sourcesJar
-}
-
-install {
-    repositories.mavenInstaller {
-        pom.project {
-            name 'Java 8 java.time JSP tags'
-            description 'JSP tag support for Java 8 java.time (JSR-310)'
-            url 'https://github.com/sargue/java-time-jsptags'
-
-            scm {
-                connection 'scm:git:git@github.com:sargue/java-time-jsptags.git'
-                developerConnection 'scm:git:git@github.com:sargue/java-time-jsptags.git'
-                url 'git@github.com:sargue/java-time-jsptags.git'
-            }
-
-            licenses {
-                license {
-                    name 'The Apache License, Version 2.0'
-                    url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
-                }
-            }
-
-            developers {
-                developer {
-                    id 'sargue'
-                    name 'Sergi Baila'
-                    email 'sargue@gmail.com'
-                }
-            }
-        }
-    }
-}
-
-bintray {
-    user = project.hasProperty('BINTRAY_USER') ? BINTRAY_USER : ''
-    key = project.hasProperty('BINTRAY_KEY') ? BINTRAY_KEY : ''
-    configurations = ['archives']
-    pkg {
-        repo = 'maven'
-        name = 'net.sargue:java-time-jsptags'
-        licenses = ['Apache-2.0']
-        vcsUrl = 'https://github.com/sargue/java-time-jsptags'
-        version {
-            name = project.version
-            desc = 'JSP tag support for Java 8 java.time (JSR-310)'
-
-            gpg {
-                sign = true
-                passphrase = project.hasProperty('BINTRAY_GPG') ? BINTRAY_GPG : ''
-            }
-        }
-    }
-}
+plugins {
+    id "com.jfrog.bintray" version "1.7.3"
+    id "java"
+    id "eclipse"
+    id "com.github.ben-manes.versions" version "0.41.0" // adds dependencyUpdates task
+    id "maven-publish"
+}
+
+group = 'net.sargue'
+archivesBaseName = 'java-time-jsptags'
+version = '2.0.0'
+
+compileJava.options.encoding = 'UTF-8'
+compileTestJava.options.encoding = 'UTF-8'
+
+java {
+    // set version of Java that the source confirms to. 
+    // The bytecode will be for this version of Java as well, unless targetCompatibility is specified.
+    sourceCompatibility = JavaVersion.VERSION_17
+}
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    // for snapshot release of spring that includes jakarta classes
+    maven {
+      url "https://repo.spring.io/milestone"
+    }    
+}
+
+dependencies {
+    implementation(group: "jakarta.servlet", name: "jakarta.servlet-api", version: "5.0.0")
+    implementation(group: "jakarta.servlet.jsp", name: "jakarta.servlet.jsp-api", version: "3.0.0")
+    implementation(group: "jakarta.servlet.jsp.jstl", name: "jakarta.servlet.jsp.jstl-api", version: "2.0.0")
```

Comment:
Why do we need to depend on the implementation of the original JSTL library?

We didn't before. Just the APIs.

Reply:
I couldn't get it to work using just the api definition.